lms/djangoapps/courseware/access.py

```diff
@@ -469,6 +469,10 @@ def _has_group_access(descriptor, user, course_key):
         # via updating the children of the split_test module.
         return ACCESS_GRANTED
 
+    # Allow staff group access, unless they are masquerading as a student.
+    if get_user_role(user, course_key) != 'student':
```

Comment:
This is the doc for the "get_user_role" method (and it does seem to match the code!):

````
    """
    Return corresponding string if user has staff, instructor or student
    course role in LMS.
    """
````

To make this code more explicit and protect against possible future rules, I will change the code to look for either staff or instructor.